src/test/java/net/minecraftforge/debug/ConfigTest.java

```diff
@@ -99,4 +133,48 @@ public SubCat(String value)
             }
         }
     }
+
+    public static class ConfigurationGui extends GuiConfig
```

Comment:
The only place mainConfigGuiClass is used is in GuiModList:285 and 418 it would be possible to edit FMLClietnHandler.getGuiFactoryFor to return a implementation that we could manually detect and initalize properly, or, we could deprecate IModGuiFactory.mainConfigGuiClass entirely and change it to a fatory method instead of doing reflection {cpw designed it reflectively for unknown reasons...}